### PR TITLE
Process.kill Function

### PR DESCRIPTION
**Pull request by briandowns**
# initial impl of Process.kill

Resolves: #694 

### What's Changed:

Adds a kill function to the Process module as well as docs.

#

### Type of Change:

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#

### Housekeeping:

- [ ] Tests have been updated to reflect the changes done within this PR (if applicable).
- [x] Documentation has been updated to reflect the changes done within this PR (if applicable).

#

### CODE REVIEW

**Comment by Jason2605** [comment]
Just checking is this still in draft?

**Comment by briandowns** [comment]
> Just checking is this still in draft?

Ready for review. Not quite sure how to test it though, effectively. I've tested it on Mac, FreeBSD, and Linux manually.

File: docs/docs/standard-lib/process.md [hunk 1]
```diff
@@ -50,3 +50,19 @@ If the external process writes to stdout and you wish to capture the output you
 Process.run(["ls", "-la"]).unwrap();
 print(Process.run(["echo", "test"], true).unwrap()); // 'test'
 ```
+
+### Process.kill(Number, Number -> Optional) -> Result\<Number>
```
Comment (Jason2605) [inline docs/docs/standard-lib/process.md:54]: This is saying it unwraps a number but it seems to be returning NIL

File: src/optionals/process.c [hunk 2]
```diff
@@ -268,6 +277,58 @@ static Value runNative(DictuVM* vm, int argCount, Value* args) {
...
+    if (argCount == 2) {
+        if (!IS_NUMBER(args[1])) {
+            runtimeError(vm, "Second argument passed to kill() must be a number");
+            return EMPTY_VAL;
+        }
+
+        signal = AS_NUMBER(args[1]);
+    }
+
+    kill(pid, signal);
```
Comment (Jason2605) [inline src/optionals/process.c:326]: Here we could do:

```suggestion
    if (kill(pid, signal) == -1) {
        ERROR_RESULT;
    }
```

Also means we can pass 0 as the signal and check if (child) processes are still running, which is a way we could test this at least on *nix.

Will have a big of a research about windows

**Comment by briandowns** [comment]
I "think" we're all set.

**Review by Jason2605 — APPROVED** [review]
Thanks for this!

**Pull request merged**